### ISSUE TITLE
About the generation of training fragments of 3d match

### ISSUE THREAD

**Issue by shinshiner** [zgojcic/3DSmoothNet#35]
Many thanks to your relatively promising work. I would like to generate some training fragments from `3d match` dataset, and I found the statement in readme that `each pair of training fragments which has more than 30% overlap (This is given in 3DMatch dataset)`. Do you mean the training pairs with more than 30% overlap are already prepared in 3d match dataset ?

I try to find such training pairs in 3d match dataset but get no result. Could you tell me where to find the training pairs? Or if I need to generate them by myself, which script in 3d match toolbox should be used?

Thanks in advance -:).

**Comment by zgojcic** [comment]
Hi, 

this sentence is indeed a bit confusing, sorry for that. 

3DMatch dataset provides the ground truth camera poses, which can be used to compute the relative transformation parameters for each pair of fragments. With these pairwise transformation the point clouds can be registered and the overlap can be computed. The official way of computing the overlap (as it is done for the test set) can be found here:

If I am not mistaken we have simply registered the point clouds and then for each point from PC1 we looked for its NN in PC2. If the distance between these two points was less than a threshold, we considered this point lying in the overlapping region. Overlap is then simply the number of points that lie in the overlapping region divided by the number of all points. As overlap can be computed in both directions PC1->PC2 and PC2->PC1 we have considered the smaller value of the two. Again if I am not mistaken, the distance threshold for the overlaping region was 0.6 cm (average resolution of the point clouds).
